src/module.c

```diff
@@ -1186,6 +1213,10 @@ int RM_CreateCommand(RedisModuleCtx *ctx, const char *name, RedisModuleCmdFunc c
     serverAssert(dictAdd(server.commands, sdsdup(declared_name), cp->rediscmd) == DICT_OK);
     serverAssert(dictAdd(server.orig_commands, sdsdup(declared_name), cp->rediscmd) == DICT_OK);
     cp->rediscmd->id = ACLGetCommandID(declared_name); /* ID used for ACL. */
+    cp->rediscmd->acl_categories = acl_categories; /* ACL categories flags for module command */
+    /* If commands have acl_categories, recompute the command bits for the existing 
+     * users to have access to these commands */  
+    if (acl_categories) ACLRecomputeCommandBitsFromCommandRulesAllUsers(); 
```

Comment:
maybe doing this per registered command is a bit harsh.
i.e. module that registers 10 commands, will re-run though all users for each command.
considering a module is only supposed to register commands when loaded, maybe we can raise some flag, and do this when the loading completed and all commands were registered.

Reply:
Yes, that sounds good!

Reply:
@MeirShpilraien @madolson i suppose that technically, till now, we allowed modules to register commands even after the OnLoad. am i right? do you see any reason to keep this possibility?

Reply:
@oranagra I do not see why we should keep (none of our modules does it).

Reply:
@oranagra I'm also fine dropping that.

Reply:
so shall we add some check in RM_CreateCommand and maybe also RM_CreateDataType and RM_Register*Config, RM_LoadConfigs that verifies we're inside the OnLoad?

Reply:
Those makes sense. The only other one I would consider is RedisModule_RegisterCommandFilter, which I assume shouldn't really be messed with at runtime either but maybe there are some use cases.

Reply:
for sure sub-commands goes together with commands.
i'm not sure about the filter, we don't really care do we?

Reply:
please describe this change (blocked APIs after init) in the PR top comment.
as well as anything else worth noting for whoever won't read the code.